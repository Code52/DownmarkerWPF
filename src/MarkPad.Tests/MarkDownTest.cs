using ApprovalTests;
using ApprovalUtilities.Utilities;
using MarkPad.Document;
using Xunit;

namespace MarkPad.Tests
{
	//[UseReporter(typeof(FileLauncherReporter))]
	public class MarkDownTest
	{
		[Fact]
		public void TestParsingBodyOnly()
		{
			var text = MarkDownWithHeader();
			var output = DocumentParser.GetBodyContents(text);
			Approvals.VerifyHtml(output);
		}
		[Fact]
		public void TestParsing()
		{
			var text = MarkDownWithHeader();
			VerifyParsing(text);
		}

		private static string MarkDownWithHeader()
		{
			var text =
				@"---
theme:TestTheme
---
# This is a header block

With a Url  [HyperLink](http://code52.org/DownmarkerWPF/) 
and some **bold** formatting and *italics*

## And a subheader

 - With a bullet point 1
 - And bullet point 2
 
 Finally there is a
 
    code { block;} =  here
    
Of course we could do `code` in the middle of the page";
			return text;
		}

		[Fact]
		public void TestParsingWithoutHeader()
		{
			VerifyParsing("This Doesn't have a header");
		}

		private static void VerifyParsing(string text)
		{
			var output = DocumentParser.Parse(text, PathUtilities.GetDirectoryForCaller());
			Approvals.VerifyHtml(output);
		}
	}
}