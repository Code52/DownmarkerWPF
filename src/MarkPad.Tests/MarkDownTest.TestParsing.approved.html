<html>
<head>
<link rel="stylesheet" type="text/css" href="TestTheme/Something.css" />

</head>
<body>
<h1>This is a header block</h1>
<p>With a Url  <a href="http://code52.org/DownmarkerWPF/" target="_blank">HyperLink</a> 
and some <strong>bold</strong> formatting and <em>italics</em></p>
<h2>And a subheader</h2>
<ul>
<li>With a bullet point 1</li>
<li>And bullet point 2</li>
</ul>
<p>Finally there is a</p>
<pre><code>code { block;} =  here
</code></pre>

<p>Of course we could do <code>code</code> in the middle of the page</p>


<script type='text/javascript'>
	window.onload = function(){
		var links = document.getElementsByTagName('a');
		for (var i = 0; i < links.length; i++) {
			var l = links[i];
			if (l.getAttribute('href')) l.target = '_blank';
		}
	};
</script>
			<script type="text/javascript" src="TestTheme/Something.js"></script>

</body>
</html>